testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/health/ConcreteHealthCheckResponseTestCase.java

```diff
@@ -0,0 +1,100 @@
+ */
+
+package org.wildfly.test.integration.microprofile.health;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.wildfly.test.integration.microprofile.health.deployment.LivenessHealthCheck;
+
+/**
+ * Validates HealthCheckResponse concrete class which was introduced in MicroProfile Health v. 2.2
+ * @author Fabio Burzigotti
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class ConcreteHealthCheckResponseTestCase {
+    private static final String DEPLOYMENT_NAME = ConcreteHealthCheckResponseTestCase.class.getSimpleName() + ".war";
+
+    @ContainerResource
+    private ManagementClient managementClient;
+
+    @Deployment
+    public static Archive<?> deploy() {
+        return ShrinkWrap.create(WebArchive.class, DEPLOYMENT_NAME)
+                .addClass(LivenessHealthCheck.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                ;
+    }
+
+    @Test
+    @Ignore("https://github.com/eclipse/microprofile-health/issues/243")
+    public void test(@ArquillianResource URL baseURL) throws IOException, URISyntaxException {
+        try (CloseableHttpClient client = HttpClientBuilder.create().build()) {
+            HttpUriRequest request = new HttpGet(
+                    String.format("http://%s:%d/health", managementClient.getMgmtAddress(), managementClient.getMgmtPort()));
+
+            try (CloseableHttpResponse response = client.execute(request)) {
+                Assert.assertEquals(
+                        "Health check endpoint call failed",
+                        HttpServletResponse.SC_OK, response.getStatusLine().getStatusCode());
+
+                final String responseContent = EntityUtils.toString(response.getEntity());
+                HealthCheckResponse typedResponse = new ObjectMapper().readValue(
```

Comment:
-1, the spec does not specify that the JSON schema must map to the `HealthCheckResponse`.